charts/redpanda/post_install_upgrade_job.go

```diff
@@ -0,0 +1,200 @@
+		will reset any values that are not specified. To work around this, we first
+		export the the configuration. If there's a change to the configuration
+	    while we're updating the exported config on disk, said change will be reverted.
+
+		TODO(chrisseto): Consolidate all cluster configuration setting to this job.
+	*/
+	script = append(script,
+		// First: dump the existing cluster configuration.
+		// We need to use config import to handle conditional configurations
+		// (e.g. cloud_storage_enabled). Maintaining a DAG of configurations
+		// is not an option for the helm chart.
+		``, ``, ``, ``, // TODO: just WS-alignment with the original template; drop these
+		`rpk cluster config export -f /tmp/cfg.yml`,
+		``, ``,
+
+		// Second: For each environment variable with the prefix RPK
+		// ("${!RPK_@}"), use `rpk redpanda config set` to update the exported
+		// config.
+
+		// Lots of Bash Jargon here:
+		//     "${KEY#*RPK_}" => Strip the RPK_ prefix from KEY.
+		//     "${config,,}" => config.toLower()
+		//     "${!KEY}" => Dynamic variable resolution. ie: What is the value of the variable with a name equal to the value of $KEY?
+
+		`for KEY in "${!RPK_@}"; do`,
+		`  config="${KEY#*RPK_}"`,
+		`  rpk redpanda config set --config /tmp/cfg.yml "${config,,}" "${!KEY}"`,
+		`done`,
+		``, ``,
+
+		// The updated file is then loaded via `+ /*rpk cluster config import*/ ` which
+		// ensures that conditional configurations (cloud_storage_enabled)
+		// "see" all their dependent keys.
+		`rpk cluster config import -f /tmp/cfg.yml`,
+		``,
+	)
+	job.Spec.Template.Spec.Containers[0].Args = append(job.Spec.Template.Spec.Containers[0].Args, unlines(script))
+
+	return job
+}
+
+// was: post-install-job-affinity
+// Set affinity for post_install_job, defaults to global affinity if not defined in post_install_job
+func postInstallJobAffinity(dot *helmette.Dot) *corev1.Affinity {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	var affinity corev1.Affinity
+	if !helmette.Empty(values.PostInstallJob.Affinity) {
+		affinity = helmette.Unwrap[corev1.Affinity](values.PostInstallJob.Affinity)
+		return &affinity
+	}
+
+	affinity.NodeAffinity = ptr.To(helmette.Unwrap[corev1.NodeAffinity](helmette.Default(map[string]any{}, values.Affinity.NodeAffinity)))
+	affinity.PodAffinity = ptr.To(helmette.Unwrap[corev1.PodAffinity](helmette.Default(map[string]any{}, values.Affinity.PodAffinity)))
+	affinity.PodAntiAffinity = ptr.To(helmette.Unwrap[corev1.PodAntiAffinity](helmette.Default(map[string]any{}, values.Affinity.PodAntiAffinity)))
+	return &affinity
+}
+
+func tolerations(dot *helmette.Dot) []corev1.Toleration {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	var result []corev1.Toleration
+	for _, t := range values.Tolerations {
+		result = append(result, helmette.Unwrap[corev1.Toleration](helmette.Default(map[string]any{}, t)))
+	}
+	return result
+}
+
+func postInstallJobResources(resources JobResources) corev1.ResourceRequirements {
```

Comment:
I have this solved in a commit I'll put up soon. I just added the corev1.ResourceRequirements to the values type.